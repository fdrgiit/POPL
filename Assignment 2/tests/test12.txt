u v w x y z #
IF THEN END DOT LPAREN RPAREN EQUAL NOTEQUAL GREATER LESS LTEQ GTEQ ID NUM REALNUM COLON SEMICOLON #
u -> IF v THEN w END #
u -> IF v THEN w DOT w END #
v -> LPAREN x RPAREN #
x -> y EQUAL y #
x -> y NOTEQUAL y #
x -> y GREATER y #
x -> y LESS y #
x -> y LTEQ y #
x -> y GTEQ y #
y -> ID #
y -> NUM #
y -> REALNUM #
w -> w z #
w -> z #
z -> ID EQUAL COLON y SEMICOLON #
##

