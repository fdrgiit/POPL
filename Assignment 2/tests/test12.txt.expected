FIRST(u) = { IF }
FIRST(v) = { LPAREN }
FIRST(w) = { ID }
FIRST(x) = { ID, NUM, REALNUM }
FIRST(y) = { ID, NUM, REALNUM }
FIRST(z) = { ID }
FOLLOW(u) = { $ }
FOLLOW(v) = { THEN }
FOLLOW(w) = { DOT, END, ID }
FOLLOW(x) = { RPAREN }
FOLLOW(y) = { EQUAL, GREATER, GTEQ, LESS, LTEQ, NOTEQUAL, RPAREN, SEMICOLON }
FOLLOW(z) = { DOT, END, ID }
